gobblin-metrics/src/main/java/gobblin/metrics/KafkaAvroReporter.java

```diff
@@ -0,0 +1,161 @@
+/* (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.metrics;
+
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.io.EncoderFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Kafka reporter for codahale metrics writing metrics in Avro format
+ *
+ * @author ibuenros
+ */
+public class KafkaAvroReporter extends KafkaReporter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KafkaAvroReporter.class);
+
+  /**
+   * Avro schema used for encoding metrics
+   * TODO: finalize metrics avro schema
+   */
+  static String SchemaString = "{\n"+
+      " \"type\": \"record\",\n"+
+      " \"name\": \"Metric\",\n"+
+      " \"namespace\":\"<TBD>\",\n"+
+      " \"fields\" : [\n"+
+      " {\"name\": \"hostname\", \"type\": [\"null\", \"string\"], \"doc\": \"hostname emitting the metric\"},\n"+
+      " {\"name\": \"environment\", \"type\": [\"null\", \"string\"], \"doc\": \"environment where metric was emitted\"},\n"+
+      " {\"name\": \"tags\", \"type\": {\"type\": \"array\", \"items\": \"string\"}, \"doc\": \"tags associated with the metric\"},\n"+
+      " {\"name\": \"name\", \"type\": \"string\", \"doc\": \"metric name\"},\n"+
+      " {\"name\": \"value\", \"type\": [\"boolean\", \"int\", \"long\", \"float\", \"double\", \"bytes\", \"string\"], \"doc\": \"metric value\"}\n"+
+      " ]\n"+
+      "}";
+
+  Schema _schema;
+  private GenericDatumWriter<GenericRecord> w;
+  private Encoder e;
+  private ByteArrayOutputStream out;
+  private int exceptionCount;
+
+  protected KafkaAvroReporter(MetricRegistry registry, String name, MetricFilter filter, TimeUnit rateUnit,
+      TimeUnit durationUnit, String brokers, String topic, String host, String env, Set<String> tags) {
+    super(registry, name, filter, rateUnit, durationUnit, brokers, topic, host, env, tags);
+
+    Schema.Parser parser = new Schema.Parser();
+    _schema = parser.parse(SchemaString);
```

Comment:
See this https://github.com/linkedin/gobblin/wiki/CodingStyle. Gobblin does not requiring starting variable names with underscore and is not doing this for all the source files.